.github/workflows/cicd.yml

```
@@ -193,27 +193,26 @@ jobs:
     strategy:
       matrix:
         os: ["ubuntu-latest"]
-        python-version: ["3.7", "3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.10", "3.11"]
         twisted-version: ["21.2", "22.1", "23.8"]
```

Maybe this also should be updated...
and add a comment to describe what rule do we use to define the targeted twisted versions ...

```suggestion
        # We have the oldest Twisted version that we support.
        # We also add the last 2 newer versions.
        twisted-version: ["21.2", "24.7", "24.10"]
```